wakuv2/waku.go

```diff
@@ -1707,6 +1707,10 @@ func (w *Waku) handleNetworkChangeFromApp(state connection.State) {
 
 func (w *Waku) ConnectionChanged(state connection.State) {
 	isOnline := !state.Offline
+	if isOnline && !w.onlineChecker.IsOnline() { // we are online and we were offline before
+		//TODO: analyze if we need to discover and connect to peers for relay.
```

Comment:
Is the `TODO` here because this PR is still in draft? Just a reminder, but you probably know there are tests for `ConnectionChanged`.